Source/WebCore/platform/unix/SharedMemoryUnix.cpp

```diff
@@ -108,29 +110,42 @@ static UnixFileDescriptor createSharedMemory()
         tempName = name.utf8();
 
         do {
+#if OS(ANDROID)
+            fileDescriptor = ASharedMemory_create(tempName.data(), size);
+#else
             fileDescriptor = shm_open(tempName.data(), O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
+#endif
         } while (fileDescriptor == -1 && errno == EINTR);
     }
 
+#if !OS(ANDROID)
     if (fileDescriptor != -1)
         shm_unlink(tempName.data());
+#endif
 #endif
 
     return UnixFileDescriptor { fileDescriptor, UnixFileDescriptor::Adopt };
 }
 
 RefPtr<SharedMemory> SharedMemory::allocate(size_t size)
 {
-    auto fileDescriptor = createSharedMemory();
+    auto fileDescriptor = createSharedMemory(size);
```

Comment:
So, we are adding ifdefs for all createSharedMemory code, why not doing something like:

```
#if OS(ANDROID)
static UnixFileDescriptor createSharedMemory(size_t size)
{
    int fileDescriptor;
    auto name = makeString("/WK2SharedMemory."_s, cryptographicallyRandomNumber<unsigned>()).utf8();
    do {
        fileDescriptor = ASharedMemory_create(name.data(), size);
    } while (fileDescriptor == -1 && errno == EINTR);

    return UnixFileDescriptor { fileDescriptor, UnixFileDescriptor::Adopt };
}
#else
static UnixFileDescriptor createSharedMemory()
{
    <current code>
}
#endif
```

Reply:
I'm assuming we don't need the 10 retries in android, but if we do, we could use the same loop here. This reduces the ifdefs and makes the code a lot easier to read without having to use a separate file for android.